src/components/PopUpWithConfirmation.js

```diff
@@ -0,0 +1,33 @@
+import PopUp from "./PopUp";
+
+export default class PopUpWithConfirmation extends PopUp {
+  constructor(popUpSelector) {
+    super({ popUpSelector });
+    this._popUpForm = this._popUpElement.querySelector(".modal__form");
+    this._submitButton = this._popUpForm.querySelector(".modal__button");
+    this._submitButtonText =
```

Comment:
Needs correction: The submit button text should be the text, not the element: `this._submitButtonText = this._submitButton.textContent`